src/Symfony/Component/Security/Core/Authentication/RememberMe/PersistentToken.php

```diff
@@ -39,7 +38,15 @@ public function __construct(string $class, string $userIdentifier, string $serie
             throw new \InvalidArgumentException('$tokenValue must not be empty.');
         }
 
-        $this->class = $class;
+        if ($tokenValue instanceof \DateTime) {
+            trigger_deprecation('symfony/security-core', '5.4', 'The $class argument of "%s" is deprecated.', __METHOD__);
```

Comment:
this class is `@internal`. We don't have to provide a BC layer. 
/cc @derrabus / @wouterj 

But you have to assert that symfony components <= 5.3 do note use this class

Reply:
I asked if I should modify it, I'm not sure about how to implement the whole PR. The scope's way bigger than I expected, I even feel like I have to modify some DoctrineBridge test about token but I'm not sure so I would like to know

Reply:
the class is declared in security-core but instantiated in security-http, so we should still provide a BC layer for the constructor IMO to simplify things.